packages/firestore/src/lite-api/aggregate.ts

```diff
@@ -15,72 +15,113 @@
  * limitations under the License.
  */
 
+import { deepEqual } from '@firebase/util';
+
 import { Value } from '../protos/firestore_proto_api';
 import { invokeRunAggregationQueryRpc } from '../remote/datastore';
 import { hardAssert } from '../util/assert';
 import { cast } from '../util/input_validation';
 
 import { getDatastore } from './components';
 import { Firestore } from './database';
-import { Query, queryEqual } from './reference';
+import { DocumentFieldValue, Query, queryEqual } from './reference';
 import { LiteUserDataWriter } from './reference_impl';
 
 /**
- * An `AggregateQuery` computes some aggregation statistics from the result set of
- * a base `Query`.
+ * An `AggregateField` computes some aggregation statistics from the result set of
+ * an aggregation query.
  */
-export class AggregateQuery {
-  readonly type = 'AggregateQuery';
-  /**
-   * The query on which you called `countQuery` in order to get this `AggregateQuery`.
-   */
-  readonly query: Query<unknown>;
+export class AggregateField<T> {
+  type = 'AggregateField';
+  _datum?: T;
```

Comment:
I found a way to get rid of this dummy `_datum` field. Please see https://github.com/dconeybe/CountApi/commit/7a8cc9a561b2ea57440a6aefa04309e69e354109 and apply that same change here.